packages/core/src/store/index.ts

```diff
@@ -33,8 +33,8 @@ const createRFStore = () =>
       return Array.from(get().nodeInternals.values());
     },
     setEdges: (edges: Edge[]) => {
-      const { defaultEdgeOptions = {} } = get();
-      set({ edges: edges.map((e) => ({ ...defaultEdgeOptions, ...e })) });
```

Comment:
@moklick this change causes a bug for folks using a redux store:
```
index.js:2820 Uncaught TypeError: Cannot add property selected, object is not extensible
    at index.js:2820:18
    at Array.reduce (<anonymous>)
    at getSelectionChanges (index.js:2817:18)
    at addSelectedEdges (index.js:3544:28)
    at onEdgeClick (index.js:1210:17)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4291:25)
    at executeDispatch (react-dom.development.js:9041:3)
```

Reverting this set state fixes the problem but i'm not sure if impacts other features implemented on this pr, i can summit a pr reverting to the old way

Reply:
this wasn't on purpose.. but it's weird that you get that error. We are not adding a property here.

Reply:
Yep i agree it's weird, this somehow triggers  [getSelectionChanges](https://github.com/wbkd/react-flow/blob/d709c7860406a691a480bcf7a81533c80287a0e5/packages/core/src/utils/changes.ts#LL132C17-L132C37) which triggers the error, just to add more context

Reply:
and you are sure that just doing:

```js
set({ edges: defaultEdgeOptions ? edges.map((e) => ({ ...defaultEdgeOptions, ...e })) : edges });
```

fixes it?

Reply:
Reverting it to `set({ edges: edges.map((e) => ({ ...defaultEdgeOptions, ...e })) });` fixes it, i can make a pr with. redux example in the vite-app tomorrow if you want to demonstrate this behaviour

Reply:
ah :/ no I get it. Of course this doesn't work. Will fix and release it today.

Reply:
nice, but i~m curious, why would this change cause such problem ?

Reply:
fixed in 11.6.1 -  redux toolkit uses immer under the hood. immer creates immutable objects and doesn't like that we do edge.selected = true / false in our code.

Reply:
Yes i got that i was just wondering why this would end up calling edge.selected = true, but no worries it's fixed, thanks for the fast response